src/solver/misc/options.cpp

```diff
@@ -45,151 +45,134 @@
 
 #include <antares/memory/memory.h>
 #include <antares/exception/AssertionError.hpp>
+#include <antares/exception/LoadingError.hpp>
 #include <antares/Enum.hpp>
+#include <antares/constants.h>
 
 #include "utils/ortools_utils.h"
 
 using namespace Yuni;
 using namespace Antares;
 using namespace Antares::Data;
 
-bool GrabOptionsFromCommandLine(int argc,
+static std::string availableOrToolsSolversString()
+{
+    const std::list<std::string> availableSolverList
+      = OrtoolsUtils().getAvailableOrtoolsSolverName();
+    std::string availableSolverListStr;
+    for (auto it = availableSolverList.begin(); it != availableSolverList.end(); it++)
+    {
+        availableSolverListStr += *it + ";";
+    }
+    // Remove last semicolumn
+    if (!availableSolverListStr.empty())
+        availableSolverListStr.pop_back();
+    return availableSolverListStr;
+}
+
+void GrabOptionsFromCommandLine(int argc,
```

Comment:
As I can understand, this function makes 2 things : 
- It defines the admissible solver command line options
- It reads the command line options the user actually chose at execution (that's the reason why we have **argc** and **argv** as an argument of this function)

We could rename this function (ex : **defineAdmissibleOptions**), remove user options parsing from it (2 last lines) and it could return the parser (note **argc** & **argv** are no more needed as arguments ). 
After the call of this function, we parse user arguments as it is done now in this function.

Reply:
This idea is great, I tried to implement it. Many thanks